pkg/envoy/proxy.go

```diff
@@ -257,7 +258,7 @@ func (p *Proxy) buildCommand(cfgPath string) *exec.Cmd {
 		p.cfg.ExtraArgs...,
 	)
 
-	cmd := exec.Command(p.cfg.ExecutablePath, args...)
+	cmd := exec.CommandContext(ctx, p.cfg.ExecutablePath, args...)
```

Comment:
Does this cancel the context after the command completes , i.e, the envoy proxy has started?
The description here makes it seem otherwise: https://pkg.go.dev/os/exec#CommandContext

Reply:
Nvm, based on your test below, looks like it does.

Reply:
Yep I think you're right @chapmanc, the goroutine should unblock and do the cleanup. Maybe worth sticking a `fmt.Println` in there and testing it manually, though?

Reply:
Confirmed after passing a logger into the test.